### ISSUE TITLE
uploader location chooser positional_accuracy conditions

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#985]
The uploader's location chooser is not quite setting `positional_accuracy` in the same way that we have previously. The main issue is that the viewport provided by the Google geocoder is not always an appropriate source of `positional_accuracy`, since it often just provides a bounding box around the most common location associated with the search terms, not the entire boundary of the place described by those search terms, e.g. searching for "yosemite national park" zooms you in to a small area around the map label for Yosemite, not to an area that contains the entire park.

The relevant existing logic can be found at https://github.com/inaturalist/inaturalist/blob/master/app/assets/javascripts/jquery/plugins/jquery.latLonSelector.js.erb#L527 and https://github.com/inaturalist/inaturalist/blob/master/app/assets/javascripts/jquery/plugins/jquery.latLonSelector.js.erb#L541

To summarize, the location chooser:

- [x] should not set `positional_accuracy` if the Google geocoder response does not specify `geometry.bounds`
- [x] should set `positional_accuracy` to be the distance from the center to one of the corners of `geometry.bounds`, e.g. the circle that circumscribes `geometry.bounds`

It might also be worth looking at those geocoder responses again and see if there's something else we can use other than `geometry.bounds`. Maybe it's changed since the last time I looked at it.

**Comment by pleary** [comment]
Pretty sure this is resolved in 9eea4cc which is on gonatus now

**Comment by kueda** [comment]
Right now searching for "yosemite national park" zooms the map to Yosemite but doesn't set the coordinates, which might be desirable in that it forces the user to choose a location, but it might give the user the mistaken impression that they've already set the coordinates when they haven't. The prior behavior was to set the coordinates without setting the positional accuracy.

Searching for something like "mount diablo state park" and then clicking on one of the options like "Mount Diablo State Park, Summit Road, Walnut Creek, CA, United States" *does* set the positional accuracy, which is also a bit dubious. Is `geometry.bounds` coming in with that geocoder result? Technically it is drawing a circle around Summit Rd on Mt. Diablo, but I can certainly imaging people choosing that option to set coordinates for the park as a whole and not just that road.

**Comment by pleary** [comment]
I'll fix the the first problem. I think I had some JS bug locally and I didn't notice it would drop a pin w/o bounds.

As for the second issue, I think part of the problem is the uploader is using a different API, the autocomplete API. If you look at https://developers.google.com/maps/documentation/javascript/places#place_search_results you'll see the kind of response it gets, and there is no `geometry.bounds` but there is a `geometry.viewport`. For searches like "yosemite national park" or "pt reyes national park" there was no `geometry.viewport` so I assumed it was a different name for `geometry.bounds`. That Mount Diablo place does include  `geometry.viewport` though, so maybe it's not exactly the same. I'm not sure what else in that response there is to work with. Here's the whole thing:

```
{
  "address_components": [
    {
      "long_name": "Mount Diablo State Park",
      "short_name": "Mount Diablo State Park",
      "types": [
        "premise"
      ]
    },
    {
      "long_name": "Summit Road",
      "short_name": "Summit Rd",
      "types": [
        "route"
      ]
    },
    {
      "long_name": "Walnut Creek",
      "short_name": "Walnut Creek",
      "types": [
        "locality",
        "political"
      ]
    },
    {
      "long_name": "Contra Costa County",
      "short_name": "Contra Costa County",
      "types": [
        "administrative_area_level_2",
        "political"
      ]
    },
    {
      "long_name": "California",
      "short_name": "CA",
      "types": [
        "administrative_area_level_1",
        "political"
      ]
    },
    {
      "long_name": "United States",
      "short_name": "US",
      "types": [
        "country",
        "political"
      ]
    },
    {
      "long_name": "94598",
      "short_name": "94598",
      "types": [
        "postal_code"
      ]
    }
  ],
  "adr_address": "<span class=\"extended-address\">Mount Diablo State Park</span>, <span class=\"street-address\">Summit Rd</span>, <span class=\"locality\">Walnut Creek</span>, <span class=\"region\">CA</span> <span class=\"postal-code\">94598</span>, <span class=\"country-name\">USA</span>",
  "formatted_address": "Mount Diablo State Park, Summit Rd, Walnut Creek, CA 94598, USA",
  "geometry": {
    "location": {
      "lat": 37.8813572,
      "lng": -121.91421029999998
    },
    "viewport": {
      "south": 37.8725499,
      "west": -121.93021770000001,
      "north": 37.8901635,
      "east": -121.8982029
    }
  },
  "icon": "https://maps.gstatic.com/mapfiles/place_api/icons/geocode-71.png",
  "id": "9c4b8b468e3eb175eaf73791867649047869b42a",
  "name": "Mount Diablo State Park",
  "place_id": "ChIJiSrkxTH0j4ARevL35v3Vljg",
  "reference": "CoQBdAAAAAfoUcfkJb8ggWY119CIkVMqIcrt6ZN7QeX2B9px3AiKtvgnpR4ky0Vkn92KBioipFze_Ou3O0SHhauNj_o-Sj4McUn1QhZilx_2g49AcBWcKUYQa3ufKrTcpgdUSYjQItWpMdj261KyK2GURId_jZToRp3mdPmEICk0xT5MiQxXEhBW28dydV4FaMazjzTOFiQ9GhTLLySHMLqEBR8i_ma4X7B8N_bV1Q",
  "scope": "GOOGLE",
  "types": [
    "premise"
  ],
  "url": "https://maps.google.com/?q=Mount+Diablo+State+Park&ftid=0x808ff431c5e42a89:0x3896d5fde6f7f27a",
  "utc_offset": -420,
  "vicinity": "Walnut Creek",
  "html_attributions": []
}
```
